server/src/user/user.service.ts

```diff
@@ -136,6 +139,51 @@ export class UserService {
     }
   }
 
+  async changePassword(userId: number, dto: ChangePasswordDto) {
+    try {
+      const user = await this.prisma.user.findUnique({
+        where: {
+          id: userId,
+        },
+      });
+
+      if (!user) {
+        throw new NotFoundException(`User with ID ${userId} not found.`);
+      }
+
+      const passwordMatch = await compare(
+        dto.currentPassword,
+        user.hashedPassword,
+      );
+
+      if (!passwordMatch) {
+        throw new UnauthorizedException('Invalid credentials');
+      }
+
+      if (dto.newPassword !== dto.confirmNewPassword) {
+        throw new BadRequestException(
+          'New password and confirmation password do not match',
+        );
+      }
```

Comment:
Oh, actually, you don't need to check this. I think if you check if the newPassword and confirmNewPassword match on the client side, you don't need to pass it to the server!

Reply:
Oh I see!
716cdb3402c6a0a80e41db82e93ae2a64a3ba0f3